src/Pages/Signup/Signup.js

```diff
@@ -1,12 +1,480 @@
 import React, { Component } from "react";
+import "./Signup.scss";
 
 class Signup extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      isModalActive: false,
+      showIdDetail: false,
+      showPwDetail: false,
+      showCheckPwDetail: false,
+      validateId: false,
+      validatePw: false,
+      validatePwLength: false,
+      isPwSame: false,
+
+      account: "",
+      password: "",
+      name: "",
+      email: "",
+      phoneNumber: "",
+      gender: "",
+      recommend: "",
+      year: "",
+      month: "",
+      day: "",
+      agree: [false, false, false, false, false, false, false],
+    };
   }
+
+  getValue = e => {
+    const { id, value } = e.target;
+    this.setState({ [id]: value });
+  };
+
+  handleModal = () => {
+    this.setState({ isModalActive: !this.state.isModalActive });
+  };
+
+  openIdDetail = () => {
+    this.setState({ showIdDetail: true });
+  };
+
+  openPwDetail = () => {
+    this.setState({ showPwDetail: true });
+  };
+
+  openCheckPwDetail = () => {
+    this.setState({ showCheckPwDetail: true });
+  };
+
+  validationId = e => {
+    const { value } = e.target;
+    const txt = /^[a-z]{6,}$/;
+    const numtxt6 = /^(?=.*[0-9])(?=.*[a-zA-Z]).{6,}$/;
+    const idValidation = txt.test(value) || numtxt6.test(value);
+    this.setState({ account: value });
+    this.setState({ validateId: idValidation ? true : false });
+  };
+
+  validationPW = e => {
+    const { value } = e.target;
+    this.setState({ password: value });
+    const txt10 = /.{10,}$/;
+    const numtxt10 = /^(?=.*[0-9])(?=.*[a-zA-Z]).{10,}$/; // 영문 + 숫자 합쳐서 10개 이상 !!!
+    const txtemoji10 = /^(?=.*[!@#$%^&*()_+])(?=.*[a-zA-Z]).{10,}$/; // 영문 + 특수기호 합쳐서 10개 이상 !!!
+    const numemoji10 = /^(?=.*[0-9])(?=.*[!@#$%^&*()_+]).{10,}$/; // 특수문자 + 숫자 합쳐서 10개 이상 !!!
+    const numtxtemoji10 = /^(?=.*[0-9])(?=.*[!@#$%^&*()_+])(?=.*[a-zA-Z]).{10,}$/; //영문+숫자+특수문자 10자 이상
+    const allCorrect =
+      numtxt10.test(value) ||
+      txtemoji10.test(value) ||
+      numemoji10.test(value) ||
+      numtxtemoji10.test(value);
+
+    if (txt10.test(value)) {
+      this.setState({ validatePwLength: true });
+    } else {
+      this.setState({ validatePwLength: false });
+    }
+    if (allCorrect) {
+      this.setState({ validatePw: true });
+    } else {
+      this.setState({ validatePw: false });
+    }
```

Comment:
이것도 if문 조건안에 들어가는 것 자체가 boolean 값이기 때문에 조건문을 굳이 안쓰셔도 될 것 같아요!
```suggestion
      this.setState({
          validatePwLength: txt10.test(value),
          validatePw: allCorrect,
        });
```

Reply:
네 수정 했습니다:)!